src/mon/commands/Command.h

```diff
@@ -0,0 +1,330 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 SUSE LLC <contact@suse.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_MON_COMMAND_H
+#define CEPH_MON_COMMAND_H
+
+#include <string>
+#include <set>
+#include <sstream>
+
+#include <boost/intrusive_ptr.hpp>
+
+#include "mon/Monitor.h"
+#include "mon/MonOpRequest.h"
+
+#include "include/types.h"
+#include "include/encoding.h"
+#include "include/ceph_assert.h"
+#include "include/Context.h"
+
+#include "common/cmdparse.h"
+#include "common/RefCountedObj.h"
+
+using namespace std;
+
+/*
+ * CommandCommon
+ *
+ * Define members and functions common to commands.
+ */
+template<typename Service>
+struct Common
```

Comment:
I like CommandBase maybe slightly more than BaseCommand, if only because it is the Base Class for Commands... but then again, it might also be read as a command /called/ Base.  This is where the idea of putting it a namespace (perhaps also consider an inline namespace) makes sense, since then it starts to matter less. +1 on nested namespace. Though, YYMV.

Also, note that your comment and the declaration no longer match. (There are a lot of opinions about how to handle this, but FWIW I lean toward just describing the thing, rather than additionally listing the name, etc.-- we're about to see that anyway. Additionally, I usually leave the closing marker of the comment connected directly above the declaration with no vertical whitespace. YYMV.)